.github/workflows/hil.yml

```diff
@@ -17,27 +17,23 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  hil:
-    name: HIL Test | ${{ matrix.target.soc }}
-    runs-on:
-      labels: [self-hosted, "${{ matrix.target.runner }}"]
+  build-tests:
```

Comment:
We could have a hil step in `ci.yaml` that only runs when the trigger is [`merge_group`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#merge_group) and that gets the artfiacts from the build and runs the tests(?)

We would delete the `hil.yml` file and basically in non-merge queue runs, it will build the tests (as we do now) and on merge queue runs it will build and execute them.

Reply:
That's good to know we can do it, but for now let's keep this simple. HIL is already way faster this way.